huggingface_pipelines/preprocessing.py

```diff
@@ -0,0 +1,201 @@
+from transformers import WhisperProcessor, WhisperFeatureExtractor
+import numpy as np
+import spacy
+from typing import List, Dict, Any, Optional
+from abc import ABC, abstractmethod
+from .pipeline import Pipeline, PipelineConfig
+import logging
+from dataclasses import dataclass
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+SPACY_MODELS = {
+    "eng_Latn": "en_core_web_sm",
+    "fra_Latn": "fr_core_news_sm",
+    "deu_Latn": "de_core_news_sm",
+    "spa_Latn": "es_core_news_sm",
+    "ita_Latn": "it_core_news_sm",
+    "por_Latn": "pt_core_news_sm",
+    "nld_Latn": "nl_core_news_sm",
+}
+
+
+@dataclass
+class PreprocessingPipelineConfig(PipelineConfig):
+    """
+    Configuration class for preprocessing pipelines.
+    Add any specific preprocessing configuration parameters here.
+    """
+    pass
+
+
+class PreprocessingPipeline(Pipeline, ABC):
+    """
+    Abstract base class for preprocessing pipelines.
+    """
+
+    def __init__(self, config: PreprocessingPipelineConfig):
+        super().__init__(config)
+
+    @abstractmethod
+    def preprocess_text(self, text: str) -> List[str]:
+        """
+        Abstract method to preprocess a single text.
+        """
+        pass
+
+    def process_batch(self, batch: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Processes a batch of data by applying preprocessing to specified columns.
+        """
+        for column in self.config.columns:
+            if column in batch:
+                batch[f"{column}_preprocessed"] = [
+                    self.preprocess_text(text) for text in batch[column]]
+
+                logger.info(
+                    f"Preprocessed {column} column:  {batch[column][:5]}")
+            else:
+                logger.warning(f"Column {column} not found in batch.")
```

Comment:
I dont think we want to see all this logs... there'll be a lot of them in normal case and they will not be informative.